support/refc/_datatypes.h

```diff
@@ -50,15 +45,58 @@ typedef struct {
   // followed by type-specific payload.
 } Value;
 
-typedef struct {
-  Value_header header;
-  uint8_t ui8;
-} Value_Bits8;
-
-typedef struct {
-  Value_header header;
-  uint16_t ui16;
-} Value_Bits16;
+/*
+We expcted at least 4bytes for Value_header alignment, to use bit0 and bit1 of
+pointer.as flags.
```

Comment:
You have couple of typos in this comment, I'd suggest something like

```diff
-We expcted at least 4bytes for Value_header alignment, to use bit0 and bit1 of
-pointer.as flags.
+We expect at least 4 bytes for `Value_header` alignment, to use bit0 and bit1 of
+pointer as flags.
```

The question is: whether is it documented that pointers are meant to be aligned and that we can rely on this? I know that in some architectures like ARM values that are used for addressing must be aligned, but it there some documented provisions of this for C pointers?